app/src/main/java/waiter/Connectable/ClientConnection.java

```diff
@@ -20,11 +21,16 @@ public ClientConnection(Socket socket) throws IOException {
     }
 
     public String read() throws IOException {
-        return this.bufferedReader.readLine();
+        StringBuilder requestContents = new StringBuilder();
+        while(!requestContents.toString().contains("\r\n")) {
```

Comment:
Not too sure if you've read [Law of Demeter](https://medium.com/vattenfall-tech/the-law-of-demeter-by-example-fd7adbf0c324), but I think the concept applies here (could be wrong).

So a lot of the newly changed code has `.` operations followed by more `.` operations. For someone like me who's terrible at reading/understanding code, this makes it really difficult to comprehend whats going on.

Perhaps you could set `requestContents.toString()` to a variable that makes me go "ah, so that's what that is" - like `formattedRequestAsString = requestContents.toString()` and then do `formattedRequestAsString.contains("\r\n")`.

Just makes for better readability, but Eli and Steven, please feel free to lmk if that doesn't apply here.

Reply:
Addressed in [this commit here](https://github.com/jonmanahan/waiter/pull/15/commits/da4cec5984fc9f2abea5b44b73e92d5d7778396c)